src/index.ts

```diff
@@ -1995,7 +1995,13 @@ export class GossipSub extends EventEmitter<GossipsubEvents> implements PubSub<G
    *
    * For messages not from us, this class uses `forwardMessage`.
    */
-  async publish(topic: TopicStr, data: Uint8Array): Promise<PublishResult> {
+  async publish(
+    topic: TopicStr,
+    data: Uint8Array,
+    publishOptions: Partial<GossipsubOpts> = {
```

Comment:
Using GossipsubOpts here is extremely broad, just allow `allowPublishToZeroPeers` and nothing else. You don't need to default to false, that what will happen anyway

Reply:
Understood. I had made it more generic in case we may want to pass other params such as `emitSelf` in the future.